lua/huez/config.lua

```diff
@@ -88,6 +105,72 @@ M.set = function(user_opts)
   M.current = vim.tbl_deep_extend("force", vim.deepcopy(M.current), user_opts)
 end
 
+---@type table<string, ThemeSetter>
+M.theme_setters = {}
+
+-- Will load all the theme configs into memory.
+M.load_theme_configs = function()
+  local theme_config_module = M.current.theme_config_module
+
+  local config_path = vim.fn.stdpath("config")
+
+  -- Convert module string to dir path string to check if dir exists
+  -- TODO: find a better way to check this
+  local theme_config_dir = config_path .. "/lua/" .. vim.fn.substitute(theme_config_module, ".", "/", "g")
+
+  -- check if config dir is non null and exists
+  -- if it does not, then just return
+  if theme_config_module == nil then
+    return
+  end
+  if vim.fn.isdirectory(theme_config_dir) == 0 then
+    logger("directory not found to load themes from", "ERROR")
+    return
+  end
+
+  local theme_manager = require("huez-manager.api")
+
+  -- if it does exist, load them into memory
+  for _, theme in pairs(theme_manager.colorscheme.installed(M.current.exclude)) do
+    ---@type boolean
+    local ok
+    ---@type Huez.ThemeConfig
+    local conf
+
+    ok, conf = pcall(require, theme_config_module .. "." .. theme)
+    -- if cannot load config, means no config found => continue
+    if not ok then
+      goto continue
+    end
+
+    logger("found configs for " .. theme, "INFO")
+    for _, theme_style in pairs(conf.styles) do
+      logger("Loading config for " .. theme .. " " .. theme_style, "INFO")
+      M.theme_setters[theme .. "-" .. theme_style] = conf.set_theme
```

Comment:
this assumes all themes a user has installed will have a `theme-style` in the namespace. Some themes like neomodern or base16 bundle many themes together that are seperate themes not denoted by a dash.

for example there's `neomodern` and `rose-prime` which are both themes contained within the same colorscheme plugin. This probably needs to be changed/handled differently

Reply:
Yeah just looked into this. Neomodern does have a different way of handling these issues. I made a fix making the `styles` component as optional and adding a default configuration entry for each theme which is the theme name itself.